pandas/tests/indexes/ranges/test_join.py

```diff
@@ -30,9 +29,9 @@ def test_join_outer(self):
             dtype=np.intp,
         )
 
-        assert isinstance(res, Int64Index)
+        assert isinstance(res, Index) and res.dtype == np.int64
```

Comment:
I found `is_int64_dtype`. IMO using that function is just a less clear way to check the same thing as I already do. Is there a benefit to using that function over checking directly?

Reply:
consistency and this would also work for Int64 for example

Reply:
Ok, I do think the other is better because we test specifically for `np.int64` here, and `pd.Int64` should give an error in this case. But not a big issue for me, I'ver updated.